components/next_steps_view/steps/create_first_channel_step/create_first_channel_step.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {createChannel} from 'mattermost-redux/actions/channels';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {Channel, ChannelType} from 'mattermost-redux/types/channels';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import {pageVisited, trackEvent} from 'actions/telemetry_actions';
+import {switchToChannel} from 'actions/views/channel';
+import {
+    setFirstChannelName,
+} from 'actions/views/channel_sidebar';
+
+import {getAnalyticsCategory} from 'components/next_steps_view/step_helpers';
+import LocalizedInput from 'components/localized_input/localized_input';
+import {StepComponentProps} from '../../steps';
+
+import {t} from 'utils/i18n';
+
+import Constants from 'utils/constants';
+import {isKeyPressed} from 'utils/utils.jsx';
+
+import {GlobalState} from 'types/store';
+
+const CreateFirstChannelStep = (props: StepComponentProps) => {
+    const [channelCreateError, setChannelCreateError] = useState(false);
+    const [buttonDisabled, setButtonDisabled] = useState(true);
+    const [channelNameValue, setChannelNameValue] = useState('');
+    const currentTeam = useSelector((state: GlobalState) => getCurrentTeam(state));
+    const userId = useSelector((state: GlobalState) => getCurrentUserId(state));
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (props.expanded) {
+            pageVisited(getAnalyticsCategory(props.isAdmin), 'pageview_create_first_channel');
+        }
+    }, [props.expanded]);
```

Comment:
Good catch, ok, so since this feature is always going to be shown to admins, there is no need of using the getAnalyticsCategory. So, no need to add the isAdmin as a hook dep and no need to call that fn.